src/H5Centry.c

```diff
@@ -503,36 +503,24 @@ H5C__flush_single_entry(H5F_t *f, H5C_cache_entry_t *entry_ptr, unsigned flags)
     if (cache_ptr->slist_enabled) {
         if (entry_ptr->in_slist) {
             assert(entry_ptr->is_dirty);
-            if (entry_ptr->flush_marker && !entry_ptr->is_dirty)
+            if (!entry_ptr->is_dirty)
                 HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, FAIL, "entry in slist failed sanity checks");
         } /* end if */
-        else {
-            assert(!entry_ptr->is_dirty);
-            assert(!entry_ptr->flush_marker);
-            if (entry_ptr->is_dirty || entry_ptr->flush_marker)
```

Comment:
Should this sanity check have been kept to ensure that there aren't entries which are NOT in the slist but ARE dirty, or is this no longer relevant?

Reply:
With this change it is possible for entries to be in dirty but not in the skip list. This happens when flushing a single object (tag) when there are dirty pieces of metadata not associated with that tag.